File: gin-enterprise-template-cron-job/internal/apiserver/biz/v1/permission/permission.go
package permission

import (
	"context"

	"github.com/clin211/gin-enterprise-template/internal/apiserver/store"
	v1 "github.com/clin211/gin-enterprise-template/pkg/api/apiserver/v1"
)

// PermissionBiz 定义处理权限请求所需的方法.
type PermissionBiz interface {
	Create(ctx context.Context, rq *v1.CreatePermissionRequest) (*v1.CreatePermissionResponse, error)
	Update(ctx context.Context, rq *v1.UpdatePermissionRequest) (*v1.UpdatePermissionResponse, error)
	Delete(ctx context.Context, rq *v1.DeletePermissionRequest) (*v1.DeletePermissionResponse, error)
	Get(ctx context.Context, rq *v1.GetPermissionRequest) (*v1.GetPermissionResponse, error)
	List(ctx context.Context, rq *v1.ListPermissionRequest) (*v1.ListPermissionResponse, error)

	PermissionExpansion
}

// PermissionExpansion 定义权限操作的扩展方法.
type PermissionExpansion interface {
	// ListPermissionTree 获取权限树
	ListPermissionTree(ctx context.Context, rq *v1.ListPermissionTreeRequest) (*v1.ListPermissionTreeResponse, error)
}

// permissionBiz 是 PermissionBiz 接口的实现.
type permissionBiz struct {
	store store.IStore
}

// 确保 permissionBiz 实现了 PermissionBiz 接口.
var _ PermissionBiz = (*permissionBiz)(nil)

func New(store store.IStore) *permissionBiz {
	return &permissionBiz{store: store}
}

File: web/src/service/api/permission.ts
import { request } from '../request';

/**
 * Get permission list
 *
 * @param params pageToken, pageSize, status, resourceType, parentID
 */
export async function fetchGetPermissionList(params?: Api.Permission.ListPermissionRequest) {
  const result = await request<Api.Permission.ListPermissionResponse>({
    url: '/v1/permissions',
    params: {
      page_token: params?.pageToken,
      page_size: params?.pageSize,
      status: params?.status,
      resource_type: params?.resourceType,
      parent_id: params?.parentID
    }
  });

  if (result.error || !result.data) {
    return result;
  }

  return {
    ...result,
    data: {
      totalCount: result.data.totalCount,
      permissions: result.data.permissions,
      pageToken: result.data.pageToken ?? ''
    } satisfies Api.Permission.ListPermissionResponse
  };
}

/**
 * Create permission
 *
 * @param data Permission data
 */
export function fetchCreatePermission(data: Api.Permission.CreatePermissionRequest) {
  return request<Api.Permission.CreatePermissionResponse>({
    url: '/v1/permissions',
    method: 'post',
    data
  });
}

/**
 * Get all permission tree
 *
 * @param params status filter
 */
export function fetchGetAllPermissionTree(params?: { status?: number }) {
  return request<Api.Permission.ListPermissionTreeResponse>({
    url: '/v1/permissions/tree',
    params
  });
}

/**
 * Get permission by ID
 *
 * @param permissionId Permission ID
 */
export function fetchGetPermission(permissionId: string) {
  return request<Api.Permission.GetPermissionResponse>({ url: `/v1/permissions/${permissionId}` });
}

/**
 * Update permission
 *
 * @param permissionId Permission ID
 * @param data Update data
 */
export function fetchUpdatePermission(permissionId: string, data: Api.Permission.UpdatePermissionRequest) {
  return request<Api.Permission.UpdatePermissionResponse>({
    url: `/v1/permissions/${permissionId}`,
    method: 'put',
    data
  });
}

/**
 * Delete permission
 *
 * @param permissionId Permission ID
 */
export function fetchDeletePermission(permissionId: string) {
  return request<Api.Permission.DeletePermissionResponse>({
    url: `/v1/permissions/${permissionId}`,
    method: 'delete'
  });
}
